Extensions/render/types.lisp

```diff
@@ -0,0 +1,36 @@
+(in-package #:mcclim-render-internals)
+
+(deftype octet ()
+  '(unsigned-byte 8))
+
+(deftype color-value ()
+  '(real 0 1))
+
+;;; Pixels
+
+(deftype argb-pixel ()
+  '(unsigned-byte 32))
+
+;;; Pixel arrays
+;;;
+;;; Pixel arrays are two-dimensional array whose dimensions are at
+;;; most 1^32-1 (which should be enough for all kinds of images and
+;;; rendering buffers while also fitting into FIXNUMs most of the
+;;; time). Elements are either 32-bit ARGB pixels for color and
+;;; opacity information or octets for stencil information.
```

Comment:
fix comment: 2^32-1 (not 1^32-1)

also, why 2^32-1? if you are after fixnum, maybe we should set it to correspond to the fixnum size? otherwise, maybe 2^30-1 would be better to take into account tagging scheme on 32 bit architectures.

Reply:
> fix comment: 2^32-1 (not 1^32-1)

Ooops.

> also, why 2^32-1? if you are after fixnum, maybe we should set it to correspond to the fixnum size? otherwise, maybe 2^30-1 would be better to take into account tagging scheme on 32 bit architectures.

I agree this is all not very principled. I also thought about helping 32 bit architectures but decided against it because I expected a "random" number like 2^30-1 to get criticized even harder :)

So 2^30-1 then?

Reply:
how about most-positive-fixnum?

(1- #.(logcount most-positive-fixnum))

Reply:
Changed.